index.js

```diff
@@ -89,59 +86,51 @@ PersonaClient.prototype.validateToken = function (token, scope, overridingScope,
-    this.redisClient.get("access_token:" + cacheKey, function (err, reply) {
-        if (reply === "OK") {
-            _this.debug("Token " + cacheKey + " verified by cache");
-            next(null, VALIDATED_TYPES.CACHE);
-        } else {
-            var requestPath = _this.config.persona_oauth_route + token;
-            var usingScope = scope || overridingScope || null;
+    this.debug("Validating token: " + token);
+    var _this = this;
 
-            if (usingScope != null) {
-                requestPath += "?scope=" + usingScope;
+    function verifyToken() {
+        var jwtConfig = {
+            algorithms: ["RS256"]
+        };
+
+        jwt.verify(token, _this.publicKey, jwtConfig, function onVerify(error, decodedToken) {
+            if(error) {
+                return next(ERROR_TYPES.VALIDATION_FAILURE, null);
+            }
+            _this.debug("Decoded token: " + decodedToken);
+            if (_.intersection(decodedToken.scopes, scopes).length > 0) {
+                next(null, VALIDATED_TYPE);
+            } else {
+                next(ERROR_TYPES.INSUFFICIENT_SCOPE, null);
             }
+        });
+    }
 
-            var options = {
-                hostname: _this.config.persona_host,
-                port: _this.config.persona_port,
-                path: requestPath,
-                method: 'HEAD'
-            };
-
-            _this.debug(JSON.stringify(options));
-
-            _this.http.request(options, function (oauthResp) {
-                if (oauthResp.statusCode === 204) {
-                    // put this key in redis with an expire
-                    _this.redisClient.multi().set("access_token:" + cacheKey, 'OK').expire("access_token:" + cacheKey, 60).exec(function (err, results) {
-                        _this.debug("cache: " + JSON.stringify(err) + JSON.stringify(results));
-                    });
-                    _this.debug("Verification passed for token " + cacheKey + ", cached for 60s");
-                    next(null, VALIDATED_TYPES.PERSONA);
-                } else {
-                    if (usingScope && usingScope !== "su") {
-                        // try su, they are allowed to perform any operation
-                        _this.validateToken (token, "su", overridingScope, next);
-                    } else {
-                        _this.debug("Verification failed for token " + cacheKey + " with status code " + oauthResp.statusCode);
-                        if (oauthResp.statusCode === 403) {
-                            next(ERROR_TYPES.INSUFFICIENT_SCOPE, null);
-                        } else {
-                            next(ERROR_TYPES.VALIDATION_FAILURE, null);
-                        }
-                    }
-                }
-            }).on ("error", function (e) {
-                _this.error("OAuth::validateToken problem: " + e.message);
-                next(e.message, null);
-            }).end();
-        }
-    });
+    if(this.publicKey == null) {
```

Comment:
I thought about fetching the public key on construction but it relies on asynchronous behaviour so it would complicate construction of the client. Here, it's just fetched on demand once if it's not yet been set.